src/EnergyPlus/PollutionModule.hh

```diff
@@ -77,336 +140,160 @@ namespace PollutionModule {
-        Real64 Coal;
-        Real64 Gasoline;
-        Real64 Propane;
-        Real64 Diesel;
-        Real64 OtherFuel1;
-        Real64 OtherFuel2;
-        Real64 ElecPurch;
-        Real64 ElecSold;
-        // Facility Meter Indexes
-        int ElecFacilityIndex;
-        int DieselFacilityIndex;
-        int PurchCoolFacilityIndex;
-        int PurchHeatWaterFacilityIndex;
-        int PurchHeatSteamFacilityIndex;
-        int NatGasFacilityIndex;
-        int GasolineFacilityIndex;
-        int CoalFacilityIndex;
-        int FuelOil1FacilityIndex;
-        int FuelOil2FacilityIndex;
-        int PropaneFacilityIndex;
-        int OtherFuel1FacilityIndex;
-        int OtherFuel2FacilityIndex;
-        int ElecProducedFacilityIndex;
-        int ElecPurchasedFacilityIndex;
-        int ElecSurplusSoldFacilityIndex;
-        // Facility Meter Values used in Pollution Calcs
-        Real64 ElecFacility;
-        Real64 DieselFacility;
-        Real64 PurchCoolFacility;
-        Real64 PurchHeatWaterFacility;
-        Real64 PurchHeatSteamFacility;
-        Real64 NatGasFacility;
-        Real64 GasolineFacility;
-        Real64 CoalFacility;
-        Real64 FuelOil1Facility;
-        Real64 FuelOil2Facility;
-        Real64 PropaneFacility;
-        Real64 OtherFuel1Facility;
-        Real64 OtherFuel2Facility;
-        Real64 ElecProducedFacility;
-        Real64 ElecPurchasedFacility;
-        Real64 ElecSurplusSoldFacility;
-
-        // Default Constructor
-        FuelTypeProps()
-            : FuelTypeNames({1, static_cast<int>(PollFactor::Num)}, Constant::eFuel::Invalid), Elec(0.0), NatGas(0.0), FuelOil1(0.0), FuelOil2(0.0),
-              Coal(0.0), Gasoline(0.0), Propane(0.0), Diesel(0.0), OtherFuel1(0.0), OtherFuel2(0.0), ElecPurch(0.0), ElecSold(0.0),
-              ElecFacilityIndex(0), DieselFacilityIndex(0), PurchCoolFacilityIndex(0), PurchHeatWaterFacilityIndex(0), PurchHeatSteamFacilityIndex(0),
-              NatGasFacilityIndex(0), GasolineFacilityIndex(0), CoalFacilityIndex(0), FuelOil1FacilityIndex(0), FuelOil2FacilityIndex(0),
-              PropaneFacilityIndex(0), OtherFuel1FacilityIndex(0), OtherFuel2FacilityIndex(0), ElecProducedFacilityIndex(0),
-              ElecPurchasedFacilityIndex(0), ElecSurplusSoldFacilityIndex(0), ElecFacility(0.0), DieselFacility(0.0), PurchCoolFacility(0.0),
-              PurchHeatWaterFacility(0.0), PurchHeatSteamFacility(0.0), NatGasFacility(0.0), GasolineFacility(0.0), CoalFacility(0.0),
-              FuelOil1Facility(0.0), FuelOil2Facility(0.0), PropaneFacility(0.0), OtherFuel1Facility(0.0), OtherFuel2Facility(0.0),
-              ElecProducedFacility(0.0), ElecPurchasedFacility(0.0), ElecSurplusSoldFacility(0.0)
-        {
-        }
+        bool used = false;
+        Real64 sourceCoeff = 0.0;
+        std::array<Real64, (int)Pollutant::Num> pollutantCoeffs = {0.0};
+        int sourceSchedNum = 0;
+        std::array<int, (int)Pollutant::Num> pollutantSchedNums = {0};
```

Comment:
This is just a few arrays instead of dozens of individual variables.

Reply:
:star_struck: